packages/millicast-sdk/src/PeerConnectionStats.js

```diff
@@ -0,0 +1,193 @@
+        /**
+        * Peer connection incoming stats.
+        *
+        * @event PeerConnection#stats
+        * @type {ConnectionStats}
+        */
+        this.emit(peerConnectionStatsEvents.stats, this.stats)
+      }
+    }, this.interval)
+  }
+
+  /**
+   * Parse incoming RTCPeerConnection stats.
+   * @param {RTCStatsReport} latestStats - RTCPeerConnection stats.
+   * @returns {ConnectionStats} RTCPeerConnection stats parsed.
+   */
+  parseStats (latestStats) {
+    const statsObject = {
+      audio: {
+        inbound: {},
+        outbound: {}
+      },
+      video: {
+        inbound: {},
+        outbound: {}
+      }
+    }
+
+    for (const report of latestStats.values()) {
+      switch (report.type) {
+        case 'outbound-rtp': {
+          const mediaType = this.getMediaType(report)
+          const codecInfo = this.getCodecData(report.codecId, latestStats)
+          const additionalData = this.getBaseReportData(report, mediaType)
+
+          const previousBytes = this.previousStats ? this.previousStats[mediaType].outbound.bytesSent : 0
+          additionalData.bitrate = 8 * (report.bytesSent - previousBytes)
+          additionalData.bytesSent = report.bytesSent
+
+          statsObject[mediaType].outbound = {
+            ...codecInfo,
+            ...additionalData
+          }
+          break
+        }
+        case 'inbound-rtp': {
+          let mediaType = this.getMediaType(report)
+          const codecInfo = this.getCodecData(report.codecId, latestStats)
+
+          // Safari is missing mediaType and kind for 'inbound-rtp'
+          if (!['audio', 'video'].includes(mediaType)) {
+            if (report.id.includes('Video')) mediaType = 'video'
+            else mediaType = 'audio'
+          }
+          const additionalData = this.getBaseReportData(report, mediaType)
+
+          const previousBytes = this.previousStats ? this.previousStats[mediaType].inbound.bytesReceived : 0
+          additionalData.bitrate = 8 * (report.bytesReceived - previousBytes)
+          additionalData.bytesReceived = report.bytesReceived
+          additionalData.packetsLost = report.packetsLost
+
+          statsObject[mediaType].inbound = {
+            ...codecInfo,
+            ...additionalData
+          }
+          break
+        }
+        default:
+          break
+      }
+    }
+
+    this.previousStats = statsObject
+    return { raw: latestStats, ...statsObject }
+  }
+
+  /**
+   * Get media type.
+   * @param {Object} report - JSON object which represents a report from RTCPeerConnection stats.
+   * @returns {String} Media type.
+   */
+  getMediaType (report) {
+    return report.mediaType || report.kind
+  }
+
+  /**
+   * Get codec information from stats.
+   * @param {String} codecReportId - Codec report ID.
+   * @param {RTCStatsReport} latestStats - RTCPeerConnection stats.
+   * @returns {Object} Object containing codec information.
+   */
+  getCodecData (codecReportId, latestStats) {
```

Comment:
Capture mimetype in codecReport
`const { mimeType } =  codecReportId ? latestStats.get(codecReportId) : {}`